### ISSUE TITLE
Cache does not clear on logout. How to compete clear the cache?

### ISSUE THREAD

**Issue by niraj-khatiwada** [apollographql/apollo-cache-persist#461]
I am using below callback method on logout by the user. But this does not clear the cache at all. Everything seems to be persisting even calling this method. How do I completely clear the cache so that on every logout, no old persisted data remains on the app. The app is in beta testing and we are getting complaints that some of the users see flash of old cards info even after logging out and logging in. 

```
 const handleLogout = React.useCallback(async () => {
    setIsLogOutLoading(true);
    try {
      if (!isInternetReachable) {
        throw new Error('No Internet');
      }
      const asyncStorageKeys = await AsyncStorage.getAllKeys();
      if (asyncStorageKeys.length > 0) {
        if (Platform.OS === 'android') {
          await AsyncStorage.clear();
        }
        if (Platform.OS === 'ios') {
          await AsyncStorage.multiRemove(asyncStorageKeys);
        }
      }
      await AsyncStorage.setItem(persistent.CAN_FORCE_LOGOUT, '0');

      await persistor.pause();
      await client.clearStore();
      await persistor.purge();
      await persistor.resume();

      await FastImage.clearDiskCache();
      await FastImage.clearMemoryCache();

      await messaging().deleteToken();
      dispatch(Logout());
    } catch (error) {
      await AsyncStorage.setItem(persistent.CAN_FORCE_LOGOUT, '1');
      onError?.(error);
    } finally {
      setIsLogOutLoading(false);
    }
  }, [dispatch, onError, client, persistor, isInternetReachable]);
```

Thank you!

**Comment by niraj-khatiwada** [comment]
Followed [this](https://github.com/apollographql/apollo-cache-persist/issues/126) issue as well but nothing seems to clear it.

**Comment by wodCZ** [comment]
Hi @niraj-khatiwada. We're using the following snippet in our app with no problems:
```ts
  await client.clearStore().catch(logger.capture); // clear client's store
  await cachePersistor.purge(); // clear cache-persist's cache
  await cachePersistor.persist(); // force write the emptied cache to storage (not sure this is needed at the moment)
```

Also, make sure that you have your other screens unmounted (you are navigated to an unauthenticated stack), so that e.g. the HomeScreen doesn't re-render and thus re-fetch the data again right after you purge the cache.

There's a `debug: true` option in cache persist options, use that to check what's going on. I believe the purging works fine, but some other part of your app is filling it back in. This would show up in logs as "Persisted cache of size 0 characters" (or 2, for `{}`) followed by another message with higher number.

**Comment by niraj-khatiwada** [comment]
> Also, make sure that you have your other screens unmounted (you are navigated to an unauthenticated stack), so that e.g. the HomeScreen doesn't re-render and thus re-fetch the data again right after you purge the cache.

But persistor.pause() should pause the persistence right? So if anything in authenticated screens re-renders, it does not persist it? Or when persistor is paused, the purge also does not work?

**Comment by wodCZ** [comment]
Yes, `pause` pauses the persistence, but you have `resume` right after the clear:
```ts
      await persistor.pause();
      await client.clearStore();
      await persistor.purge();
      await persistor.resume();
```
This all probably executes within 10ms. Even if you resumed later, on an unauthenticated screen, the `client` could already perform few queries since your `clearStore`, and those would get persisted (which is a correct behaviour).

The only way to ensure you don't have any leftover data in your cache is to make sure no query happens after your `client.clearStore()` call (or that they fail, therefore don't get cached).

An apollo link such as [apollo-link-logger](https://github.com/xoredg/apollo-link-logger) could give you a better insight on what's happening — I can't help much more without more details.

**Comment by niraj-khatiwada** [comment]
So I changed the code to this:
```
      await persistor.pause();
      await client.clearStore();
      await persistor.purge();
      await persistor.resume();
      await persistor.persist();
```

It seems to purge and persist only size 2 ("{}").
![1](https://user-images.githubusercontent.com/51436631/168603930-9a777941-92b0-45a7-bb8f-cd47e537bd4a.png)

After that it logs out and there are no additional appolo-cache-logs.

Now after loggin in it the logs are this:
But why does it say persisted 157095, the old size. Those were the cache size from multiple query in multiple screens. So how is it persisting that again?
![2](https://user-images.githubusercontent.com/51436631/168604012-12dec1f1-ef6b-4427-8e46-26b8d125bf78.png)

**Comment by niraj-khatiwada** [comment]
@wodCZ do we have log for pause and resume? because I am not seeing it and the data is still persisted when I call persist after pause. Why is that?

**Comment by wodCZ** [comment]
Np, good luck with your launch 😉

**Comment by niraj-khatiwada** [comment]
> Np, good luck with your launch wink

Thank you so much.
